docs/DeveloperGuide.md

```diff
@@ -1,38 +1,180 @@
 # Developer Guide
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
+
+### Architecture
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+Given below is a quick overview of main components and how they interact with each other.
+
+<img src="images/ArchitectureDiagram.png" width="300" />
+
+**Main components of the architecture**
+
+## **Implementation**
+
+### Add transaction feature
+The add transaction feature is facilitated by the `Parser` class which parses user input and creates a `Expense` or
+`Allowance` object which extends from the `Transaction` class. The created `Transaction` object will be stored in a 
+`TransactionList`.
+
+Given below is an example usage scenario and how the add transaction mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `TransactionList` will be initialized.
+
+Step 2. The user executes `expense /amt 20 /date 24-10-2023 /desc Lunch /note Pasta /cat Food` command to create a
+transaction. The `expense` command calls `Parser#parseExpense()` to create an `Expense` object. The 
+`AddCommand#execute()` is then called to store the `Expense` object in the `TransactionList`.
+
+The following sequence diagram shows how the add transaction operation works:
+
+<img src="images/AddTransactionSequenceDiagram.png" width="600" />
+
+### View transaction feature
+
+#### I. Architecture-Level Design
+The "View Transaction" feature primarily involves the following components:
+
+1. Parser: Responsible for interpreting user input and generating a ViewCommand object.
+
+2. ViewCommand: A subclass of the Command class, created by the Parser to represent the "view" command.
+
+3. Nuscents: The main application class that receives and executes commands. It invokes the execute() method of the ViewCommand.
+
+4. TransactionList: A data structure to store and manage transactions.
+
+5. Transaction: Represents individual transactions.
+
+6. UI: Handles user interface and messaging.
+
+#### II. Component-Level Design
+1. Parser
+
+The Parser class identifies the "view" command and extracts the taskIndex (transaction index) from the user's input.
+
+2. ViewCommand
+
+The ViewCommand object is created by the Parser. It encapsulates the user's request to view a specific transaction. This object is passed to the Nuscents class for execution.
+
+3. Nuscents
+
+In the Nuscents class, the execute() method of the ViewCommand is called, and the taskIndex is extracted from the command. It then calls the viewTransaction(taskIndex) method on the TransactionList.
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/2ae0d837-1595-47bc-9705-87407a091498)

The numbering is broken in the github.io site